### PR TITLE
Use new e2etemplates

### PR DESCRIPTION
**Pull request by MarcelMue**
towards https://github.com/giantswarm/giantswarm/issues/4388

This uses the e2etemplates for `kvm-operator` and issues unique names for `clusterroles` and `clusterrolebindings`.

### CODE REVIEW

File: integration/setup/setup.go [hunk 1]
```diff
@@ -98,7 +99,27 @@ func Resources(g *framework.Guest, h *framework.Host) error {
...
+		{
+			c := chartvalues.KVMOperatorConfig{
+				ClusterName: env.ClusterID(),
+				ClusterRole: chartvalues.KVMOperatorClusterRole{
+					BindingName: fmt.Sprintf("%s-kvm-operator", env.ClusterID()),
+					Name:        fmt.Sprintf("%s-kvm-operator", env.ClusterID()),
+				},
+				ClusterRolePSP: chartvalues.KVMOperatorClusterRole{
+					BindingName: fmt.Sprintf("%s-kvm-operator-psp", env.ClusterID()),
+					Name:        fmt.Sprintf("%s-kvm-operator-psp", env.ClusterID()),
```
Comment (xh3b4sd) [inline integration/setup/setup.go]: For the future it would be cool to put these things in the `env` package or lets just have a `key` package.

**Review by xh3b4sd — APPROVED** [review]
Nice man.

File: integration/setup/setup.go (hunk 1 above)
Comment (MarcelMue) [reply integration/setup/setup.go]: I'll do that in a follow up with other operators, I think the env file makes sense for now.

File: integration/setup/setup.go [hunk 2]
```diff
@@ -98,7 +99,27 @@ func Resources(g *framework.Guest, h *framework.Host) error {
 			return microerror.Mask(err)
 		}
 
-		err = h.InstallBranchOperator("kvm-operator", "kvmconfig", template.KVMOperatorChartValues)
```
Comment (kopiczko) [inline integration/setup/setup.go]: Should `template.KVMOperatorChartValues` be removed from the project?

Comment (MarcelMue) [reply integration/setup/setup.go]: It is removed but is listed in the changes below as a `move` from template folder to the vendor. So it's moved but not expanded due to now being in the vendor.

Github being github :man_shrugging:

**Review by kopiczko — APPROVED** [review]

**Pull request merged**